The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11350425100/job/31568800535) [(plain)](https://github.com/rust-lang/rust/commit/bbe5dde7ddf9f29a76312da2b921223f6e9b6442/checks/31568800535/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 2.933 Building wheels for collected packages: reuse
#16 2.934   Building wheel for reuse (pyproject.toml): started
#16 3.180   Building wheel for reuse (pyproject.toml): finished with status 'done'
#16 3.182   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#16 3.182   Stored in directory: /tmp/pip-ephem-wheel-cache-egwp5qov/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#16 3.184 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#16 3.582 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#16 3.583 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#16 DONE 3.7s
---
    Checking addr2line v0.22.0
error[E0425]: cannot find value `Release` in this scope
  --> std/src/sys/sync/once_box.rs:63:62
   |
63 |         match self.ptr.compare_exchange(null_mut(), new_ptr, Release, Acquire) {
   |
help: consider importing one of these unit variants
   |
8  + use crate::sync::atomic::Ordering::Release;
8  + use crate::sync::atomic::Ordering::Release;
   |
8  + use core::sync::atomic::Ordering::Release;
   |

error: unused import: `AcqRel`
  --> std/src/sys/sync/once_box.rs:11:37
   |
11 | use crate::sync::atomic::Ordering::{AcqRel, Acquire, Relaxed};
   |
   = note: `-D unused-imports` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(unused_imports)]`

```

</details>

--- a/library/std/src/sys/sync/once_box.rs
+++ b/library/std/src/sys/sync/once_box.rs
@@ -8,7 +8,7 @@
 use crate::mem::replace;
 use crate::ptr::null_mut;
 use crate::sync::atomic::AtomicPtr;
-use crate::sync::atomic::Ordering::{AcqRel, Acquire, Relaxed};
+use crate::sync::atomic::Ordering::{Acquire, Relaxed, Release};
 
 pub(crate) struct OnceBox<T> {
     ptr: AtomicPtr<T>,
@@ -60,7 +60,7 @@ impl<T> OnceBox<T> {
     #[cold]
     fn initialize(&self, f: impl FnOnce() -> Box<T>) -> &T {
         let new_ptr = Box::into_raw(f());
-        match self.ptr.compare_exchange(null_mut(), new_ptr, AcqRel, Acquire) {
+        match self.ptr.compare_exchange(null_mut(), new_ptr, Release, Acquire) {
             Ok(_) => unsafe { &*new_ptr },
             Err(ptr) => {
                 // Lost the race to another thread.

